prov/efa/src/rdm/efa_rdm_ep.h

```diff
@@ -443,4 +443,47 @@ ssize_t efa_rdm_ep_post_handshake(struct efa_rdm_ep *ep, struct efa_rdm_peer *pe
 void efa_rdm_ep_post_handshake_or_queue(struct efa_rdm_ep *ep,
 				     struct efa_rdm_peer *peer);
 
+static inline int efa_rdm_attempt_to_sync_memops(struct efa_rdm_ep *ep, void *buf, void *desc)
+{
+	int err = 0;
+	struct efa_mr *efa_mr = (struct efa_mr *) desc;
+
+	if (OFI_UNLIKELY(ep->cuda_api_permitted && efa_mr && efa_mr->needs_sync)) {
+		err = cuda_set_sync_memops(buf);
+		if (err) {
+			EFA_WARN(FI_LOG_MR, "Unable to set memops for cuda ptr\n");
+			return err;
+		}
+		efa_mr->needs_sync = false;
+	}
+
+	return err;
+}
+
+static inline int efa_rdm_attempt_to_sync_memops_iov(struct efa_rdm_ep *ep, struct iovec *iov, void **desc, int num_desc)
+{
```

Comment:
maybe need another check for `desc` itself before you dereferencing `desc[i]`

probably unnecessary because passing `desc` as NULL and a non-zero `num_desc` is illegal

Reply:
Apparently IntelMPI has a bug :)
```
(gdb) bt
#0  0x00007f5d684e7c31 in efa_rdm_attempt_to_sync_memops_iov (ep=0x150a730, iov=0x7f5dab0470e0 <MPIDI_OFI_global+47648>, desc=0x0, num_desc=1)
    at ./prov/efa/src/rdm/efa_rdm_ep.h:468
#1  0x00007f5d684ea999 in efa_rdm_msg_generic_recv (ep=0x150a730, msg=0x7f5dab047160 <MPIDI_OFI_global+47776>, tag=0, ignore=0, op=0, flags=16842752)
    at prov/efa/src/rdm/efa_rdm_msg.c:954
#2  0x00007f5d684eab85 in efa_rdm_msg_recvmsg (ep_fid=0x150a730, msg=0x7f5dab047160 <MPIDI_OFI_global+47776>, flags=16842752)
    at prov/efa/src/rdm/efa_rdm_msg.c:1003
#3  0x00007f5daa35114c in MPIDI_OFI_mpi_init_hook (rank=22062896, size=-1425772320, appnum=0, tag_bits=0x1, init_comm=0x0)
    at /p/pdsd/scratch/Uploads/IMPI/other/software/libfabric/linux/v1.9.0/include/rdma/fi_endpoint.h:284
#4  0x00007f5da9f00bf7 in MPID_Init (requested=22062896, provided=0x7f5dab0470e0 <MPIDI_OFI_global+47648>) at ../../src/mpid/ch4/src/ch4_init.c:1546
#5  0x00007f5daa1c7590 in MPIR_Init_thread (argc=0x150a730, argv=0x7f5dab0470e0 <MPIDI_OFI_global+47648>, user_required=0, provided=0x1)
    at ../../src/mpi/init/initthread.c:175
#6  0x00007f5daa1c6e4b in PMPI_Init (argc=0x150a730, argc@entry=0x7ffe3ea32f9c, argv=0x7f5dab0470e0 <MPIDI_OFI_global+47648>, 
    argv@entry=0x7ffe3ea32f90) at ../../src/mpi/init/init.c:139
#7  0x00000000004018e4 in main (argc=<optimized out>, argv=<optimized out>) at osu_bw.c:61
```

Reply:
Let's add a check for `desc` itself now for safety